### PR TITLE
Add Ui message for load failure and also add new test for saving

### LINKED ISSUE #59: MVP - Save feature (load)
When loading fails due to corrupted file, denied access or other unexpected I/O errors: display corresponding error message (and create new empty address book, which is already implemented).
_**refer to feature specifications to standardize outputs_

### PR DESCRIPTION
**Pull request by PrishaVP**
fixes #59

### CODE REVIEW

File: src/test/java/seedu/address/storage/JsonAddressBookStorageTest.java [hunk 1]
```diff
@@ -106,4 +109,10 @@ private void saveAddressBook(ReadOnlyAddressBook addressBook, String filePath) {
     public void saveAddressBook_nullFilePath_throwsNullPointerException() {
         assertThrows(NullPointerException.class, () -> saveAddressBook(new AddressBook(), null));
     }
+
+    @Test
+    public void saveAddressBook_writeAccessDenied_throwsAccessDeniedException() {
```
Comment (PrishaVP) [inline src/test/java/seedu/address/storage/JsonAddressBookStorageTest.java:114]:
```suggestion
    public void saveAddressBook_writeAccessDenied_throwsAccessDeniedException() {
    Path filePath = Path.of("writeAccessDeniedAddressBook.json");
        File file = filePath.toFile();  // Convert Path to File

        if (file.setWritable( false,
                false)) {
            System.out.println("File cannot be written to");
        }

```

File: src/test/java/seedu/address/storage/JsonAddressBookStorageTest.java [hunk 2]
```diff
@@ -106,4 +110,18 @@ private void saveAddressBook(ReadOnlyAddressBook addressBook, String filePath) {
...
+    Path filePath = Path.of("writeAccessDeniedAddressBook.json");
```
Comment (PrishaVP) [inline src/test/java/seedu/address/storage/JsonAddressBookStorageTest.java:116]:
```suggestion
        Path filePath = Path.of("writeAccessDeniedAddressBook.json");
```

File: src/test/java/seedu/address/storage/JsonAddressBookStorageTest.java [hunk 3]
```diff
@@ -94,10 +96,12 @@ public void saveAddressBook_nullAddressBook_throwsNullPointerException() {
...
         try {
-            new JsonAddressBookStorage(Paths.get(filePath))
-                    .saveAddressBook(addressBook, addToTestDataPathIfNotNull(filePath));
+            JsonAddressBookStorage jsonAddressBook = new JsonAddressBookStorage(Paths.get(filePath));
+            jsonAddressBook.saveAddressBook(addressBook, addToTestDataPathIfNotNull(filePath));
+        } catch (AccessDeniedException e) {
+            throw new AccessDeniedException("File write access denied");
         } catch (IOException ioe) {
             throw new AssertionError("There should not be an error writing to the file.", ioe);
         }
```
Comment (PrishaVP) [inline src/test/java/seedu/address/storage/JsonAddressBookStorageTest.java:107]:
```suggestion
        }
        
```

File: src/test/java/seedu/address/storage/JsonAddressBookStorageTest.java [hunk 4]
```diff
@@ -94,13 +96,16 @@ public void saveAddressBook_nullAddressBook_throwsNullPointerException() {
...
+
```
Comment (PrishaVP) [inline src/test/java/seedu/address/storage/JsonAddressBookStorageTest.java:108]:
```suggestion
```

File: src/test/data/JsonAddressBookStorageTest/notJsonFormatAddressBook.json [hunk 5]
```diff
@@ -1 +1 @@
-not json format!
+not json format!
```
Comment (PrishaVP) [inline src/test/data/JsonAddressBookStorageTest/notJsonFormatAddressBook.json:1]:
```suggestion
not json format!

```

**Review by wkxcass — COMMENTED** [review]
Just a few nits!

File: src/main/java/seedu/address/MainApp.java [hunk 6]
```diff
@@ -45,7 +47,11 @@ public class MainApp extends Application {
     protected Storage storage;
     protected Model model;
     protected Config config;
+    private boolean isLoadError;
 
+    public boolean getIsLoadError() {
+        return isLoadError;
+    }
```
Comment (PrishaVP) [inline src/main/java/seedu/address/MainApp.java:54]:
```suggestion
    }
    
```

File: src/main/java/seedu/address/ui/MainWindow.java [hunk 7]
```diff
@@ -70,6 +74,23 @@ public MainWindow(Stage primaryStage, Logic logic) {
...
+     * Displays a message if there is a data loading error.
+     *
+     * @param model the model that is used
+     */
+    public static CommandResult displayDataLoadErrorMessage(Model model) {
+        requireNonNull(model);
+        CommandResult commandResult = new CommandResult("");
+
+        if (model.getAddressBook().equals(new AddressBook())) {
+            commandResult = new CommandResult("Unable to load contacts due to unexpected data loading error!");
```
Comment (PrishaVP) [inline src/main/java/seedu/address/ui/MainWindow.java:87]:
```suggestion
            commandResult = new CommandResult("Unable to load contacts due to unexpected data loading error");
```

File: src/main/java/seedu/address/MainApp.java [hunk 8]
```diff
@@ -45,7 +47,12 @@ public class MainApp extends Application {
...
+
```
Comment (PrishaVP) [inline src/main/java/seedu/address/MainApp.java:55]:
```suggestion
```

**Comment by wkxcass** [comment]
LGTM!

**Pull request merged**